src/main/scala/io/github/pauljamescleary/petstore/domain/users/UserService.scala

```diff
@@ -39,7 +39,7 @@ class UserService[F[_]](userRepo: UserRepositoryAlgebra[F], validation: UserVali
   def update(user: User)(implicit M: Monad[F]): EitherT[F, UserNotFoundError.type, User] =
     for {
       _ <- validation.exists(user.id)
-      saved <- EitherT.liftF(userRepo.put(user))
+      saved <- EitherT.fromOptionF(userRepo.update(user), UserNotFoundError)
```

Comment:
nice!

Reply:
yeah that transformation fell out nicely.

Reply:
I applied the same trick in several places in 7a14f63